src/diffusers/pipelines/musicldm/pipeline_musicldm.py

```diff
@@ -391,6 +397,40 @@ def prepare_latents(self, batch_size, num_channels_latents, height, dtype, devic
         latents = latents * self.scheduler.init_noise_sigma
         return latents
 
+    def enable_model_cpu_offload(self, gpu_id=0):
```

Comment:
Ok for now, but why not use the default way of model offloading here?

Reply:
Something I noticed while working on this.

Certain pipelines (AudioLDM2, MusicLDM, Shap E) do not make use of the forward method of their components. Instead they pass inputs into submodules of the component. 
https://github.com/huggingface/diffusers/blob/dfec61f4b3797bb140b570456ff44a00d7f52502/src/diffusers/pipelines/musicldm/pipeline_musicldm.py#L189

This leads to a device mismatch error since accelerate only moves the module back to GPU when forward is called.
```
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu...
```

IMO, if Pipelines are using submodules of the components during inference, I think it's fine for them to implement their own  enable_model_cpu_offload since it can be challenging for us to know exactly which modules to offload.

Reply:
In this PR, I've cleaned up the enable_model_cpu_offload in the problematic pipelines to properly offload the submodules so that users still get the expected memory savings. Alternatively, we could move these problematic modules into the _exclude_from_cpu_offload list and use the enable_model_cpu_offload defined in DiffusionPipeline but that would affect memory savings.